File: Classes/LED.m
#import "LED.h"

@implementation LED

- (void) toggleTorch
{
	Class captureDeviceClass = NSClassFromString(@"AVCaptureDevice");
	if (captureDeviceClass != nil) {
		AVCaptureDevice *device = [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo];
		if ([device hasTorch]) {
			[device lockForConfiguration:nil];
			[device setTorchModeOnWithLevel:.1 error:nil];
			[device unlockForConfiguration];
		}
		[device lockForConfiguration:nil];
		[device setTorchMode:AVCaptureTorchModeOff];
		[device unlockForConfiguration];
	}
}

@end

File: Classes/LED.swift
import AVFoundation

class LED {

    static let device = AVCaptureDevice.default(for: AVMediaType.video)

    class func flash() {
        let device = LED.device
        if device!.isTorchAvailable {
            DispatchQueue.main.async(execute: {
                do {
                    try device!.lockForConfiguration()
                } catch _ {
                }
                do {
                    try device!.setTorchModeOn(level: 0.1)
                } catch _ {
                }
                device!.torchMode = .off
                device!.unlockForConfiguration()
            })

        }
    }

    class func hasLED() -> Bool {
        if LED.device != nil {
            return LED.device!.hasFlash
        }
        else {
            return false
        }
    }

}
